### PR TITLE
Fix/skale 2801 nonce too low

### PR DESCRIPTION
**Pull request by badrogger**

### CODE REVIEW

**Review by dmytrotkk — CHANGES_REQUESTED** [review]
tests failed

**Comment by dmytrotkk** [comment]
Tests fixed, nonce works fine now 🙃

**Review by dmytrotkk — APPROVED** [review]

File: main.py [hunk 1]
```diff
@@ -57,17 +58,18 @@ def _sign_and_send():
...
-        logger.info(f'Signing stansaction with {nonce_manager.nonce}')
-        try:
-            tx = wallet.sign_and_send(transaction_dict)
-        except Exception as e:  # todo: catch specific error
-            logger.error('Error occured', exc_info=e)
-            return construct_err_response(HTTPStatus.BAD_REQUEST, e)
-        nonce_manager.increment()
-        logger.info(f'Transaction sent - tx: {tx}, nonce: {transaction_dict["nonce"]}')
-        return construct_ok_response({'transaction_hash': tx})
+    logger.info(f'thread_id: {threading.get_ident()}, waiting for the lock')
```
Comment (badrogger) [inline main.py]: Why context manager was removed ?

File: tests/server_test.py [hunk 2]
```diff
@@ -85,4 +89,37 @@ def test_sign_hash(skale_bp):
...
+            'transaction_dict': tx_dict_str
+        })
+        assert isinstance(data['transaction_hash'], str)
+
+
+def test_multhreading(skale_bp):
+    threads = []
+    N_THREADS = 5
+    for i in range(0, N_THREADS):
+        thread = CustomThread(
```
Comment (badrogger) [inline tests/server_test.py:113]: I think using regular Thread will be enough in this case

File: main.py (hunk 1 above)
Comment (dmytrotkk) [reply main.py]: returned context manager

**Pull request merged**